files/helpers.py

```diff
@@ -371,7 +373,7 @@ def media_file_info(input_file):
         "filename": input_file,
         "file_size": file_size,
         "video_duration": video_duration,
-        "video_frame_rate": float(Fraction(video_info["r_frame_rate"])),
+        "video_frame_rate": Fraction(video_info["r_frame_rate"]),
```

Comment:
This indeed makes files.helpers.media_file_info return something that is not JSON serializable and thus 

```
self.media_info = json.dumps(ret)
```

breaks on files.models.set_media_type

So if this is set to int, this won't break. But what is the problem you mention with cast to float?

Reply:
Well, if you have a framerate that is actually a periodical number, this won't work and yield errors at latest when you try to re-encode specifying the (now rounded) framerate.

I think the most elegant way would be to store the framerate as numerator and denominator and use that, or keep the framerate as-is, except for cases where it's really really high.